lib/ansible/modules/cloud/vmware/vmware_host_sriov.py

```diff
@@ -0,0 +1,331 @@
+            if pci_device.id == self._getPciId(host):
+                try:
+                    if pci_device.sriovCapable:
+                        pnic_info["sriovCapable"] = True
+                        pnic_info["sriovEnabled"] = pci_device.sriovEnabled
+                        pnic_info["sriovActive"] = pci_device.sriovActive
+                        pnic_info["numVirtualFunction"] = pci_device.numVirtualFunction
+                        pnic_info[
+                            "numVirtualFunctionRequested"
+                        ] = pci_device.numVirtualFunctionRequested
+                        pnic_info[
+                            "maxVirtualFunctionSupported"
+                        ] = pci_device.maxVirtualFunctionSupported
+                    else:
+                        pnic_info["sriovCapable"] = False
+                except AttributeError:
+                    pnic_info["sriovCapable"] = False
+                break
+        return pnic_info
+
+    def _getPciId(self, host):
+        for pnic in host.config.network.pnic:
+            if pnic.device == self.vmnic:
+                return pnic.pci
+        self.module.fail_json(msg="No nic= %s on host= %s" % (self.vmnic, host.name))
+
+    def _update_sriov(self, host, sriovEnabled=None, numVirtualFunction=None):
+        if sriovEnabled is None and numVirtualFunction is None:
+            self.results["changes"][host.name][
+                "msg"
+            ] = "No any changes, already configured"
+            return False
+        nic_sriov = vim.host.SriovConfig()
+        nic_sriov.id = self._getPciId(host)
+        if sriovEnabled is not None:
+            nic_sriov.sriovEnabled = sriovEnabled
+            self.results["changes"][host.name]["sriovEnabled"] = sriovEnabled
+        if numVirtualFunction is not None and numVirtualFunction >= 0:
+            nic_sriov.numVirtualFunction = numVirtualFunction
+            self.results["changes"][host.name][
+                "numVirtualFunction"
+            ] = numVirtualFunction
+
+        try:
+            if not self.module.check_mode:
+                host.configManager.pciPassthruSystem.UpdatePassthruConfig([nic_sriov])
+                # looks only for refresh info
+                host.configManager.pciPassthruSystem.Refresh()
+                sleep(2)  # TODO: needed method to know that host updated info
+                return True
+            return False
+        except vim.fault.HostConfigFault as config_fault:
+            self.module.fail_json(
+                msg="Failed to configure SR-IOV for host= %s due to : %s"
+                % (host.name, to_native(config_fault.msg))
+            )
+            return False
+
+
+def main():
+    """Main"""
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        cluster_name=dict(type="str", required=False),
+        esxi_hostname=dict(type="str", required=False),
```

Comment:
cluster_name or esxi_hostname any one of them should be required.

Reply:
yes, but not both of them in a same time.
this check done in
```
module = AnsibleModule(
        argument_spec=argument_spec,
        required_one_of=[["cluster_name", "esxi_hostname"]],
```
in
`./ansible/lib/ansible/module_utils/common/validation.py`
```
def check_required_one_of(terms, module_parameters):
    """Check each list of terms to ensure at least one exists in the given module
    parameters

    Accepts a list of lists or tuples
```

Reply:
What I meant is one of them should be made required=True. 
What happens if user doesnt pass both of them?

Reply:
> What happens if user doesnt pass both of them?

all will be fine
```
vic@AN: ~/ansible vmware_host_sriov
$ cat ./t_args.json                                                                                      [14:25:08]
{
        "ANSIBLE_MODULE_ARGS": {
                "hostname": "10.72.24.143",
                "username": "administrator@vsphere.local", 
                "password": "Aff", 
                "validate_certs": false, 
                "sriov_on": true,
                "num_virt_func": 8, 
                "vmnic": "vmnic7"
        }
}

(envPy37_ansible_dev)  
vic@AN: ~/ansible vmware_host_sriov
$ python -m ansible.modules.cloud.vmware.vmware_host_sriov ./t_args.json                                 [14:25:30]

{"msg": "one of the following is required: cluster_name, esxi_hostname", "failed": true, "invocation": {"module_args": {"hostname": "10.72.24.143", "username": "administrator@vsphere.local", "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", "validate_certs": false, "sriov_on": true, "num_virt_func": 8, "vmnic": "vmnic7", "port": 443}}}
(envPy37_ansible_dev) FAIL: 1

vic@AN: ~/ansible vmware_host_sriov
```